modules/c++/sys/source/OSUnix.cpp

```diff
@@ -346,6 +373,39 @@ void sys::OSUnix::getMemInfo(size_t &totalPhysMem, size_t &freePhysMem) const
 #endif
 }
 
+std::string sys::OSUnix::getCurrentExecutable(
+        const std::string& argvPathname) const
+{
+    std::vector<std::string> possibleSymlinks;
+
+    // Linux
+    possibleSymlinks.push_back(sys::Path::joinPaths(
+            sys::Path::delimiter()[0] + std::string("proc"),
+            sys::Path::joinPaths("self", "exe")));
+
+    // Solaris
+    possibleSymlinks.push_back(sys::Path::joinPaths(
+            sys::Path::delimiter()[0] + std::string("proc"),
+            sys::Path::joinPaths("self",
+            sys::Path::joinPaths("path", "a.out"))));
+
+    for (size_t ii = 0; ii < possibleSymlinks.size(); ++ii)
+    {
+        struct stat linkStatus;
+        const std::string pathname = possibleSymlinks[ii];
+        linkStat(pathname, linkStatus);
```

Comment:
Ah, so I was originally trying to use that to get the size of path. But then the stat.st_size was 0, which I figured had something to do with the target being the running program. Anyway, it’s pulled out now completely.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Wednesday, July 20, 2016 8:19 PM
To: mdaus/coda-oss <coda-oss@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [mdaus/coda-oss] Current path (#146)


In modules/c++/sys/source/OSUnix.cpp<https://github.com/mdaus/coda-oss/pull/146#discussion_r71629791>:

> +    // Linux

> +    possibleSymlinks.push_back(sys::Path::joinPaths(

> +            sys::Path::delimiter()[0] + std::string("proc"),

> +            sys::Path::joinPaths("self", "exe")));

> +

> +    // Solaris

> +    possibleSymlinks.push_back(sys::Path::joinPaths(

> +            sys::Path::delimiter()[0] + std::string("proc"),

> +            sys::Path::joinPaths("self",

> +            sys::Path::joinPaths("path", "a.out"))));

> +

> +    for (size_t ii = 0; ii < possibleSymlinks.size(); ++ii)

> +    {

> +        struct stat linkStatus;

> +        const std::string pathname = possibleSymlinks[ii];

> +        linkStat(pathname, linkStatus);

So actually... why is linkStat() called? Nothing does anything with linkStatus. Did a check or some other usage get removed in some of the refactoring?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/mdaus/coda-oss/pull/146/files/33ce58c5a69eebcb9e29bf331d9ce7fbe5b75f44#r71629791>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFRAYMbEa9C2enhV5cVVMHFgh13iIlwyks5qXrr5gaJpZM4JLdyF>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.